src/Writer/ErrorLog.php

```diff
@@ -0,0 +1,100 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zend-log for the canonical source repository
+ * @copyright Copyright (c) 2005-2016 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+
+namespace Zend\Log\Writer;
+
+use Traversable;
+use Zend\Log\Formatter\Simple as SimpleFormatter;
+use Zend\Validator\EmailAddress as EmailAddressValidator;
+
+class ErrorLog extends AbstractWriter
+{
+    /**
+     * @var int
+     */
+    protected $mode = 0;
+
+    /**
+     * @var string|null
+     */
+    protected $destination = null;
+
+
+    /**
+     * ErrorLog constructor.
+     * @param array|null|\Traversable $options
+     */
+    public function __construct($options = [])
+    {
+        if ($options instanceof Traversable) {
+            $options = iterator_to_array($options);
+        }
+
+        if (! is_array($options)) {
+            $options = [];
+        }
+
+        parent::__construct($options);
+
+        $mode = isset($options['mode']) ? (int) $options['mode'] : 0;
+        if (0 <= $mode && $mode <= 4) {
+            $this->mode = $mode;
+        }
+
+        $destination = isset($options['destination']) ? (string) $options['destination'] : null;
+        $is_stream = $this->mode == 3 && $this->isStream($destination) ;
+
```

Comment:
Please throw an exception when:

- mode is 3, and
- destination is not a valid stream

As this indicates invalid configuration.